torch/csrc/utils/python_dispatch.cpp

```diff
@@ -73,6 +75,72 @@ inline torch::CppFunction dispatch_str(const char* key, Func&& raw_f) {
   }
 }
 
+class PythonKernelHolder : public c10::OperatorKernel {
+  PyObject* func_;
+
+public:
+
+  PythonKernelHolder(py::object func) : func_(func.release().ptr()) {}
+  // This is a generally useful pattern and safer than directly using pybind11's
+  // py::object destructor.  This is because this object may outlive
+  // libtorch_python, so we want to disarm the deallocation if that happens.
+  // PyInterpreter does this correctly, pybind11 does not.
+  ~PythonKernelHolder() {
+    getPyInterpreter()->decref(func_);
```

Comment:
Is that a pattern we should start using everywhere instead of checking if python is initialized before taking the GIL and doing the decref?

Reply:
If you store a Python object in a C++ class that can be used in non-Python contexts, yes, this is the safest way to do it. (Python side data structures that are only ever deallocated by Python refcounter/gc you do not need this)